WinUIGallery/Samples/SamplePages/TabViewWindowingSamplePage.xaml.cs

```diff
@@ -85,64 +137,174 @@ private void Tabs_ExternalTornOutTabsDropping(TabView sender, TabViewExternalTor
 
     private void Tabs_ExternalTornOutTabsDropped(TabView sender, TabViewExternalTornOutTabsDroppedEventArgs args)
     {
-        int position = 0;
-
-        foreach (TabViewItem tab in args.Tabs.Cast<TabViewItem>())
-        {
-            GetParentTabView(tab)?.TabItems.Remove(tab);
-            sender.TabItems.Insert(args.DropIndex + position, tab);
-            position++;
-        }
+        MoveDataItems(GetTabItemDataList(tabTearOutWindow), TabItemDataList, args.Items, args.DropIndex);
     }
 
-    private TabView GetParentTabView(TabViewItem tab)
+    private static Window CreateNewWindow()
     {
-        DependencyObject current = tab;
+        var newPage = new TabViewWindowingSamplePage();
+        var window = WindowHelper.CreateWindow();
+        window.SystemBackdrop = new MicaBackdrop();
+        window.ExtendsContentIntoTitleBar = true;
+        window.Content = newPage;
+        window.AppWindow.SetIcon("Assets/Tiles/GalleryIcon.ico");
+        newPage.SetupWindowMinSize(window);
+        return window;
+    }
 
-        while (current != null)
+    private static void MoveDataItems(ObservableCollection<TabItemData> source, ObservableCollection<TabItemData> destination, object[] dataItems, int index)
+    {
+        foreach (object tabItemData in dataItems)
         {
-            if (current is TabView tabView)
-            {
-                return tabView;
-            }
+            source.Remove((TabItemData)tabItemData);
+            destination.Insert(index, (TabItemData)tabItemData);
```

Comment:
@sabareesanms This is where the app is crashing for me, trying to insert at index 1 while destination is empty. The repro is to move the last tab of a window out and you should be able to this the crash.